packages/@aws-cdk/aws-sagemaker/test/test.endpoint.ts

```diff
@@ -0,0 +1,170 @@
+                Tags: [
+                    { Key: "Name", Value: "Endpoint" }
+                ]
+            }));
+
+            test.done();
+        },
+        "with many properties set, it correctly creates the endpoint resources"(test: Test) {
+            // GIVEN
+            const region = 'test-region'; // hardcode the region
+            const stack = new cdk.Stack(undefined, undefined, {
+                env: {
+                    region
+                },
+            });
+            const containerImage = new sagemaker.GenericContainerDefinition();
+            // create the notebook instance
+            const model = new sagemaker.Model(stack, 'Model', {
+                primaryContainer: containerImage,
+            });
+            const kmsKey = new kms.Key(stack, "Key");
+
+            // WHEN
+            const endpoint = new sagemaker.Endpoint(stack, 'Endpoint', {
+                configName: "myconfig",
+                productionVariants: [
+                    {
+                        model,
+                        variantName: "myvariant",
+                        initialInstanceCount: 10,
+                        initialVariantWeight: 20,
+                        instanceType: new ec2.InstanceType('p3.2xlarge'),
+                    }
+                ],
+                kmsKey,
+            });
+            endpoint.node.applyAspect(new cdk.Tag("Project", "myproject"));
+
+            // THEN
+            expect(stack).to(haveResource("AWS::SageMaker::EndpointConfig", {
+                EndpointConfigName: "myconfig",
+                ProductionVariants: [
+                    {
+                        InitialInstanceCount: 10,
+                        InitialVariantWeight: 20,
+                        InstanceType: 'ml.p3.2xlarge',
+                        ModelName: { "Fn::GetAtt": [ "Model2AD80A05", "ModelName" ]},
+                        VariantName: "myvariant",
+                    }
+                ],
+                KmsKeyId: { "Fn::GetAtt": [ "Key961B73FD", "Arn" ]  }
+            }));
+            expect(stack).to(haveResource("AWS::SageMaker::Endpoint", {
+                EndpointConfigName: { "Fn::GetAtt": [ "EndpointEndpointConfig5871F635", "EndpointConfigName" ] },
+                Tags: [
+                    { Key: "Name", Value: "Endpoint" },
+                    { Key: "Project", Value: "myproject" },
+                ]
+            }));
+
+            test.done();
+        },
+        "it throws error when no production variant given"(test: Test) {
+            // GIVEN
+            const stack = new cdk.Stack();
+
+            // WHEN
+            const endpoint = new sagemaker.Endpoint(stack, 'Endpoint');
+
+            const errors = validate(stack);
+
+            test.equal(errors.length, 1);
+            const error = errors[0];
+            test.same(error.source, endpoint);
+            test.equal(error.message, "Must have at least one Production Variant");
```

Comment:
You can simplify this with something like:

```ts
test.deepEqual(errors.map(e => e.message), ["Must have at least one Production Variant"]);
```

Same comment for all tests dealing with error checking :)